ceno_zkvm/src/tables/ram/ram_impl.rs

```diff
@@ -415,11 +362,177 @@ impl<DVRAM: DynVolatileRamTable + Send + Sync + Clone> DynVolatileRamTableConfig
                         config.addr,
                         DVRAM::addr(&config.params, i) as u64
                     );
-                    *structural_row.last_mut().unwrap() = F::ONE;
+                    if i < num_instances {
+                        *structural_row.last_mut().unwrap() = F::ONE;
+                    }
                 });
             Ok([RowMajorMatrix::empty(), structural_witness])
         }
     }
+
+    /// support offset address and length from public io
```

Comment:
nitpick: we'd better distinguish `public io` vs `public values` as `public io` is a subset of the vm's `public values`. PV includes VM's running info like `init_pc / end_pc`. Therefore, we should use `public values` here.